src/components/molecules/InternalShadowRectButton/InternalShadowRectButton.tsx

```diff
@@ -143,11 +147,34 @@ const StyledButtonWrapper = styled(OakBox)<
  *  `onHovered?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, duration: number) => void;`<br>
  *  called after a mouseEnter and mouseLeave event has happened
  */
+
+function handleButtonSize(size: ButtonSizes): {
+  pv: SpacingStyleProps["$pv"];
+  ph: SpacingStyleProps["$ph"];
+  font: TypographyStyleProps["$font"];
+} {
+  switch (size) {
+    case "small":
+      return {
+        pv: "inner-padding-xs",
+        ph: "inner-padding-s",
+        font: "body-3-bold",
+      };
+    default:
+      return {
+        pv: "inner-padding-s",
+        ph: "inner-padding-m",
+        font: "heading-7",
+      };
+  }
+}
```

Comment:
This looks a lot like a variant.

I tried to get this sort of thing past @kimon-satan way back and i'm reasonably certain he will tap a sign and point to the design heuristics https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design in particular

"Fewer props more components"

In this case I think he'd see this as a new opinionated component:

```typescript
function OakSmallPrimaryButton(props) {
  return (
    <OakPrimaryButton
      pv="inner-padding-xs"
      ph="inner-padding-s"
      font="body-3-bold"
      {...props}
    />;
  );
}
```

That said, is there a design for this https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?m=auto&t=utBQRsEXibp2C1h0-1 ?

Reply:
Ah I did think weather this should be a component... The adding of a handler did make me question weather logic should be at this level. I'll re implement following your advice. Thank you @carlmw !!

p.s. I am unsure weather Anna has added it to the design system yet, it is in process though. The button is coming out of designs that were used for the ai-beta pre this alignment. It can be seen in the labs.thenational.academy/aila